### ISSUE TITLE
Thread safety issue with FI

### ISSUE THREAD

**Issue by Tomas-Kraus** [eclipse-ee4j/jaxb-fi#28]
A CXF user is running into a thread safety issue when using FI. It MAY
actually be a JDK bug though. You can see the full stack trace at:
[https://issues.apache.org/jira/secure/attachment/12424481/FullStackTrace.log](https://issues.apache.org/jira/secure/attachment/12424481/FullStackTrace.log)
but the gist is that there are SEVERAL threads like:

java.lang.Thread.State: RUNNABLE
at java.lang.String.intern(Native Method)
at
com.sun.xml.fastinfoset.util.NamespaceContextImplementation.getNamespaceURI(Nam
at
org.apache.cxf.interceptor.WrappedOutInterceptor.handleMessage(WrappedOutInterc

Basically, stuck in String.intern called from
NamespaceContextImplementation.getNamespaceURI
#### Environment
Operating System: All
Platform: All
URL: [https://issues.apache.org/jira/browse/CXF-2530](https://issues.apache.org/jira/browse/CXF-2530)
#### Affected Versions
[current]
